Sources_v3/Workers/MessageUpdater_Tests.swift

```diff
@@ -0,0 +1,559 @@
+        }
+        
+        super.tearDown()
+    }
+    
+    // MARK: Edit message
+    
+    func test_editMessage_propogates_CurrentUserDoesNotExist_Error() throws {
+        // Simulate `editMessage(messageId:, text:)` call
+        var completionCalledError: Error?
+        messageUpdater.editMessage(messageId: .unique, text: .unique) {
+            completionCalledError = $0
+        }
+        
+        // Assert `MessageDoesNotExist` is received
+        AssertAsync.willBeTrue(completionCalledError is ClientError.CurrentUserDoesNotExist)
+    }
+    
+    func test_editMessage_propogates_MessageDoesNotExist_Error() throws {
+        // Create current user is the database
+        try database.createCurrentUser()
+        
+        // Simulate `editMessage(messageId:, text:)` call
+        var completionCalledError: Error?
+        messageUpdater.editMessage(messageId: .unique, text: .unique) {
+            completionCalledError = $0
+        }
+        
+        // Assert `MessageDoesNotExist` is received
+        AssertAsync.willBeTrue(completionCalledError is ClientError.MessageDoesNotExist)
+    }
+    
+    func test_editMessage_propogates_MessageCanNotBeUpdatedByCurrentUser_Error() throws {
+        let anotherUserMessagePayload: MessagePayload<DefaultDataTypes> = .dummy(
+            messageId: .unique,
+            authorUserId: .unique
+        )
+        
+        // Create current user and channel in the database
+        let (_, cid) = try createCurrentUserAndChannel()
+
+        // Create message authored by another user in the database
+        try database.writeSynchronously { session in
+            try session.saveMessage(payload: anotherUserMessagePayload, for: cid)
+        }
+        
+        // Try to edit another user's message
+        var completionCalledError: Error?
+        messageUpdater.editMessage(messageId: anotherUserMessagePayload.id, text: .unique) {
+            completionCalledError = $0
+        }
+        
+        // Assert `MessageCanNotBeUpdatedByCurrentUser` is received
+        AssertAsync.willBeTrue(completionCalledError is ClientError.MessageCanNotBeUpdatedByCurrentUser)
+    }
+
+    func test_editMessage_updatesLocalMessageCorrectly_ifThereIsNoLocalState() throws {
+        let updatedText: String = .unique
+
+        // Create current user and channel in the database
+        let (_, cid) = try createCurrentUserAndChannel()
+        
+        // Create a new message in the database
+        let createdMessageId = try createMessage(in: cid, localState: nil)
+
+        // Edit created message with new text
+        var completionCalled = false
+        messageUpdater.editMessage(messageId: createdMessageId, text: updatedText) {
+            XCTAssertNil($0)
+            completionCalled = true
+        }
+        
+        // Load created message
+        var message: MessageDTO? {
+            database.viewContext.message(id: createdMessageId)
+        }
```

Comment:
I realized this is an unnecessary pattern when working with DTOs. We used to have it like this for the models. For DTOs you can simply do:
```swift
let message = database.viewContext.message(id: createdMessageId)
```

only for models it's required to do the following, since models are immutable:
```swift
var messageModel: Message? { database.viewContext.message(id: createdMessageId)?.asModel() }
```

Reply:
I can imagine one case where this pattern can be useful: when we are waiting for a DTO to be created and check it asynchronously in tests. But that's not the case here 😄